wfe/web-front-end.go

```diff
@@ -87,8 +87,10 @@ func statusCodeFromError(err interface{}) int {
 		return http.StatusForbidden
 	case core.NotFoundError:
 		return http.StatusNotFound
+	case core.LengthRequiredError:
+		return http.StatusLengthRequired
 	case core.SignatureValidationError:
-		return http.StatusPreconditionFailed
+		return http.StatusBadRequest
```

Comment:
In case you wonder what this was about: StatusBadRequest was always the right response code for SignatureValidationError. However, the handlers below were hardcoding StatusBadRequest, so this bug didn't manifest. When I switched them to use statusCodeFromError, the tests failed. So I had to also fix this case. See also #595, better error handling in WFE.